core/src/main/scala/org/apache/spark/deploy/RRunner.scala

```diff
@@ -100,15 +100,17 @@ object RRunner {
         builder.redirectErrorStream(true) // Ugly but needed for stdout and stderr to synchronize
         val process = builder.start()
 
-        new RedirectThread(process.getInputStream, System.out, "redirect R output").start()
+        val stdoutBuffer = new CircularBuffer(1024)
```

Comment:
how large is each flush? can it be bigger than 1024 and causing some content to be lost (overwritten)?